### PR TITLE
restore point create and delete API

### PR DESCRIPTION
**Pull request by sarvesh-s**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by sarvesh-s** [comment]
Rest spec links
https://github.com/Azure/azure-rest-api-specs-pr/pull/374
https://github.com/Azure/azure-rest-api-specs-pr/pull/392
https://github.com/Azure/azure-rest-api-specs/pull/2593

**Review by jaredmoo — CHANGES_REQUESTED** [review]
Need test for delete, otherwise looks good :)

File: src/SDKs/_metadata/sql_resource-manager.txt [hunk 1]
```diff
@@ -1,11 +1,13 @@
-2018-01-31 10:51:15 UTC
+
+Type dsc to start/install the DS Consolidated Console
+
+2018-03-09 17:32:10 UTC
 
 1) azure-rest-api-specs repository information
-GitHub user: kosta-bizetic
+GitHub user: Azure
```
Comment (shahabhijeet) [inline src/SDKs/_metadata/sql_resource-manager.txt]: @sarvesh-s please pull from upstream and regenerate using generate.cmd
It has the fix to get you the commit id for your generation

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks like there are some breaking changes introduced here. Please bump the package version to `1.13.0-preview` in the csproj file.
Please pull down the latest changes from upstream and run
`msbuild build.proj /t:build /p:Scope=SDKs\SqlManagement`
This should update the `.props` file. Please commit these changes.

**Comment by shahabhijeet** [comment]
@sarvesh-s we will do a final review as soon as @jaredmoo approves this PR

**Comment by shahabhijeet** [comment]
@sarvesh-s this PR needs your attention. If you are not able to move forward, I request you to close this PR and open it when it's ready to be reviewed and signed off in due to time by your team.
This PR will be closed by EOW

**Review by jaredmoo — APPROVED** [review]

**Comment by sarvesh-s** [comment]
I updated the PR, now I only have a test change. Rest of the code was already merged as part of different PR, @shahabhijeet

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Any idea why
src/SDKs/_metadata/sql_resource-manager.txt 
is a binary file in your PR?

**Comment by sarvesh-s** [comment]
@jaredmoo Any idea why its binary file shown here as I don't see it locally, its plain text for me

**Comment by jaredmoo** [comment]
```
git fetch --all
git reset --hard jaredmoo/sarvesh-fix
```

Then force push

**Pull request closed without merging**